File: packages/zynth-document-picker/ios/ZynthDocumentPicker.swift
import Foundation
import ZynthKit

@objc(ZynthDocumentPicker)
public class ZynthDocumentPicker: NSObject {
  private static var moduleInstance: DocumentPickerModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthDocumentPicker] Module already initialized")
      return
    }

    let module = DocumentPickerModule(runtime: runtime)
    runtime.installModules([module])
    moduleInstance = module
    print("[ZynthDocumentPicker] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}

File: packages/zynth-document-picker/android/ZynthDocumentPicker/src/main/java/dev/zynth/documentpicker/ZynthDocumentPicker.kt
package dev.zynth.documentpicker

import android.util.Log
import androidx.activity.ComponentActivity
import androidx.activity.result.contract.ActivityResultContracts
import com.zynth.kit.runtime.ZynthRuntime

object ZynthDocumentPicker {
    private const val TAG = "ZynthDocumentPicker"

    @JvmStatic
    fun initialize(activity: ComponentActivity, runtime: ZynthRuntime) {
        Log.d(TAG, "Initializing ZynthDocumentPicker")

        val module = DocumentPickerModule(runtime, activity)

        module.openDocumentLauncher =
            activity.registerForActivityResult(ActivityResultContracts.OpenDocument()) { uri ->
                module.onSingleDocumentResult(uri)
            }

        module.openMultipleDocumentsLauncher =
            activity.registerForActivityResult(ActivityResultContracts.OpenMultipleDocuments()) { uris ->
                module.onMultipleDocumentsResult(uris)
            }

        runtime.installModules(listOf(module))
        Log.d(TAG, "ZynthDocumentPicker initialized")
    }
}
